api/src/main/java/io/onetable/spi/extractor/SourceClient.java

```diff
@@ -48,6 +49,14 @@ public interface SourceClient<COMMIT> extends Closeable {
    */
   SchemaCatalog getSchemaCatalog(OneTable table, COMMIT commit);
```

Comment:
Do you think we can eliminate the `SchemaCatalog` class with this change?
It was one of the objectives of #77